model/graph/graph.go

```diff
@@ -0,0 +1,88 @@
+package graph
+
+import (
+	"errors"
+
+	"github.com/sankethkini/FamilyTreeInGo/model/node"
+)
+
+var NodeNotFoundErr = errors.New("node not found")
+
+type IGraph interface {
+	AddNode(id, name string) node.INode
+	RemoveNode(id string) error
+	AllNodes() []node.INode
```

Comment:
why is `AllNodes()` returning an interface, ideally it should return array of Node struct

Reply:
same for other functions